### PR TITLE
Gh/service bus

### PR DESCRIPTION
**Pull request by geneh**

### CODE REVIEW

File: providers/queuing/serviceBusQueue.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+      metrics[this.name][operation].incr();
+    }
+  }
+
+  _setLockRenewalTimer(request, attempts = 0, delay = 4.5 * 60 * 1000) {
+    attempts++;
+    const timeoutId = setTimeout(() => {
+      this.client.renewLockForMessage(request._message, (renewLockError) => {
+        if (renewLockError) {
+          this.logger.error(renewLockError);
```
Comment (jeffmcaffer) [inline providers/queuing/serviceBusQueue.js]: should this retry? return?

File: providers/queuing/serviceBusQueue.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+  abandon(request) {
+    const deferred = Q.defer();
+    this.client.unlockMessage(request._message, (error) => {
+      if (error) {
+        return deferred.reject(error);
+      }
+      this._incrementMetric('abandon');
+      this._log('NAKed', request);
+      clearTimeout(request._timeoutId);
```
Comment (jeffmcaffer) [inline providers/queuing/serviceBusQueue.js]: consider putting the timeout on the _message . this is more of a message level thing tha na request thing. Technically the upper levels could recreate the request

File: providers/queuing/serviceBusFactory.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
```
Comment (jeffmcaffer) [inline providers/queuing/serviceBusFactory.js]: update these to be the same as the rest of GHCrawler

Comment (geneh) [reply providers/queuing/serviceBusFactory.js]: As discussed, we'll leave it for now.

**Pull request merged**